backend/src/room/room.gateway.ts

```diff
@@ -0,0 +1,68 @@
+import { UseGuards } from "@nestjs/common";
+import {
+  ConnectedSocket,
+  MessageBody,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from "@nestjs/websockets";
+import { InjectPinoLogger, PinoLogger } from "nestjs-pino";
+import { Namespace, Socket } from "socket.io";
+
+import { WsAuthGuard } from "src/auth/ws.guard";
+
+import { RoomService } from "./room.service";
+
+import { ROOM_EVENTS, ROOM_NAMESPACE } from "~shared/constants";
+
+@UseGuards(WsAuthGuard)
+@WebSocketGateway({ namespace: ROOM_NAMESPACE })
+export class RoomGateway {
```

Comment:
Since we are using `handleDisconnect`, let's make the gateway implement the OnGatewayDisconnect interface.
(ref: https://docs.nestjs.com/websockets/gateways#lifecycle-hooks)

Same for the `matchGateway`.

Reply:
For matchGateway, we are passing the difficulty level as message so that we can delete from redis. But `handleDisconnect` only takes in socket as the param